clients/java/src/main/java/io/camunda/zeebe/client/impl/worker/JobWorkerImpl.java

```diff
@@ -114,7 +116,7 @@ public boolean isOpen() {
 
   @Override
   public boolean isClosed() {
-    return !isOpen() && claimableJobPoller.get() != null && remainingJobs.get() <= 0;
+    return !isOpen() && claimableJobPoller.get() != null && jobsQueue.isEmpty();
```

Comment:
I know this is out of scope, but now I wonder, if the executor is closed, the worker will never close? :shrug: